File: Modules/HumanResource/PayrollSettings/PayrollSettings/PayrollSettingsForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.PayrollSettings")]
[BasedOnRow(typeof(PayrollSettingsRow), CheckNames = true)]
public class PayrollSettingsForm
{
    /*
    public DateTime EffectiveFrom { get; set; }
    public DateTime EffectiveUntil { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public short IsActive { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
    */
    [Category("Seperation Bonus")]
    [QuarterWidth]
    public bool SeperateBonus { get; set; }
    [QuarterWidth]
    public bool AnnualizedBonus { get; set; }
    [Category("Seperation Incentive")]

    [QuarterWidth]
    public bool SeperateIncentive { get; set; }
    [QuarterWidth]
    public bool AnnualizedIncentive { get; set; }

    //[QuarterWidth]
    //public bool SeperateMoneyClaiming { get; set; }
   // [QuarterWidth]
   // public bool SeperateAllowance { get; set; }
    [Category("Bonus Subjections")]
    public bool BonusSubjectEpf { get; set; }
    public bool BonusSubjectSocso { get; set; }
    public bool BonusSubjectEis { get; set; }
    public bool BonusSubjectHrdf { get; set; }
    public bool BonusSubjectPcb { get; set; }

    [Category("Incentive Subjections")]
    public bool IncentiveSubjectEpf { get; set; }
    public bool IncentiveSubjectSocso { get; set; }
    public bool IncentiveSubjectEis { get; set; }
    public bool IncentiveSubjectHrdf { get; set; }
    public bool IncentiveSubjectPcb { get; set; }


}

File: Modules/ServerTypes/PayrollSettings/PayrollSettingsForm.ts
import { BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PayrollSettingsForm {
    SeperateBonus: BooleanEditor;
    AnnualizedBonus: BooleanEditor;
    SeperateIncentive: BooleanEditor;
    AnnualizedIncentive: BooleanEditor;
    BonusSubjectEpf: BooleanEditor;
    BonusSubjectSocso: BooleanEditor;
    BonusSubjectEis: BooleanEditor;
    BonusSubjectHrdf: BooleanEditor;
    BonusSubjectPcb: BooleanEditor;
    IncentiveSubjectEpf: BooleanEditor;
    IncentiveSubjectSocso: BooleanEditor;
    IncentiveSubjectEis: BooleanEditor;
    IncentiveSubjectHrdf: BooleanEditor;
    IncentiveSubjectPcb: BooleanEditor;
}

export class PayrollSettingsForm extends PrefixedContext {
    static formKey = 'PayrollSettings.PayrollSettings';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PayrollSettingsForm.init)  {
            PayrollSettingsForm.init = true;

            var w0 = BooleanEditor;

            initFormType(PayrollSettingsForm, [
                'SeperateBonus', w0,
                'AnnualizedBonus', w0,
                'SeperateIncentive', w0,
                'AnnualizedIncentive', w0,
                'BonusSubjectEpf', w0,
                'BonusSubjectSocso', w0,
                'BonusSubjectEis', w0,
                'BonusSubjectHrdf', w0,
                'BonusSubjectPcb', w0,
                'IncentiveSubjectEpf', w0,
                'IncentiveSubjectSocso', w0,
                'IncentiveSubjectEis', w0,
                'IncentiveSubjectHrdf', w0,
                'IncentiveSubjectPcb', w0
            ]);
        }
    }
}